lib/card/set_pattern.rb

```diff
@@ -77,6 +77,15 @@ def initialize card
         else
           Card.fetch_id @anchor_name
         end
+        if self.class.pattern_code == 'type' and
+             default_rule = Card.fetch(card.type_cardname.trait_name(:type).trait_name(:default),
+                                       :skip_modules=>true, :skip_virtual=>true) and
+             type_code = default_rule.type_code and
+             mod_key = "Type::#{type_code.to_s.camelize}" and
+             (Card::Set.modules[:nonbase_format].values + [Card::Set.modules[:nonbase]]).
+               any?{|hash| hash[mod_key]}
+          @inherited_type_module_key = mod_key
```

Comment:
Both great comments.  I was thinking that I wanted to put this in the set pattern file for type, but the standard there is all class methods.  Overriding it here will work great.  And I'll have to check out how the rule_cache gets used.